### DISCUSSION TITLE
Rust, Slint app from MacOS to ChromeOS Flex

### DISCUSSION THREAD

**Discussion by csavels** [slint-ui/slint#4951] [Q&A]
Hello,

Dev OS: Darwin, MacOS (M2)
Target: Last ChromeOS Flex
Slint: 1.5.1
Rust: 1.75.0

I would like to compile my application (which includes Slint UI) to run on a computer powered by ChromeOS Flex.

I've tried 
`cargo build --target=x86_64-unknown-linux-gnu --release`

- The application cannot be used on ChromeOS
- The application runs via "Penguin", but Slint doesn't work.

Error via Penguin: 

> "Could not initialize backend. Error from Winit backend: Error init winit event loop: the requested operation is not supported by Wini. No backends configured".

Could you advise me how to resolve this error or recommend a solution to make the application compatible directly on ChromeOS Flex? 

Thanks in advance.

**Comment by tronical** [comment]
I don’t think anyone has tried this combination yet :-). It looks like winit can’t load wayland. Should apps under Chrome OS Flex use Wayland?

Could you try cloning https://github.com/rust-windowing/winit/ and try running `cargo run —example window`?

**Reply by csavels** [reply]
Thanks for your reply :)
Apparently ChromeOS Flex does use wayland.
I don't know why, but the application now launches, but I've done absolutely nothing new and I'm trying on the same file as before

**Reply by dfaure-kdab** [reply]
2 years later: I'm using slint on wayland every day, works fine.
